Mathlib/Order/Partition/Basic.lean

```diff
@@ -0,0 +1,118 @@
+/-
+Copyright (c) 2025 Peter Nelson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Peter Nelson
+-/
+import Mathlib.Data.SetLike.Basic
+import Mathlib.Order.SupIndep
+
+/-!
+# Partitions
+
+A `Partition` of an element `a` in a complete lattice is an independent family of nontrivial
+elements whose supremum is `a`.
+
+An important special case is where `s : Set α`, where a `Partition s` corresponds to a partition
+of the elements of `s` into a family of nonempty sets.
+This is equivalent to a transitive and symmetric binary relation `r : α → α → Prop`
+where `s` is the set of all `x` for which `r x x`.
+
+## Main definitions
+
+* For `[CompleteLattice α]` and `s : α`, a `Set.Partition s` is an independent collection of
+  nontrivial elements whose supremum is `s`.
+
```

Comment:
Can we have TODOs to 
a) Link this to the finpartition version
b) Give helper lemmas for the (common) set case
(for the latter, things like restating sSup_parts_eq in terms of set unions, and `le_of_mem` as `subset_of_mem` are examples - I'm happy for them to be in other PRs rather than here but they should be in mathlib eventually)